Framework/Kernel/inc/MantidKernel/VMD.h

```diff
@@ -234,12 +234,12 @@ template <typename TYPE = double> class DLLExport VMDBase {
     boost::split(strs, str, boost::is_any_of(", "));
 
     std::vector<TYPE> vals;
-    for (size_t d = 0; d < strs.size(); d++) {
-      if (!strs[d].empty()) {
+    for (const auto &str : strs) {
+      if (!str.empty()) {
         TYPE v;
-        if (!Strings::convert(strs[d], v))
+        if (!Strings::convert(str, v))
           throw std::invalid_argument(
-              "VMDBase: Unable to convert the string '" + strs[d] +
+              "VMDBase: Unable to convert the string '" + str +
               "' to a number.");
         vals.push_back(v);
       }
```

Comment:
Since ` ` is a valid separator, I don' think we need `StringTokenizer::TOK_TRIM`.

Since data is already a pointer to the first element, we can just pass `data` instead of `&data[0]`.